exchanges/order/orders.go

```diff
@@ -475,6 +476,62 @@ func (s *Submit) DeriveSubmitResponse(orderID string) (*SubmitResponse, error) {
 	}, nil
 }
 
+// AdjustBaseAmount will adjust the base amount of a submit response if the
+// exchange has modified the amount. This is usually due to decimal place
+// restrictions or rounding. This will return an error if the amount is zero
+// or the submit response is nil.
+func (s *SubmitResponse) AdjustBaseAmount(a float64) error {
+	if s == nil {
+		return errOrderSubmitResponseIsNil
+	}
+
+	if a <= 0 {
+		return errAmountIsZero
+	}
+
+	if s.Amount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "exchange %v has adjusted requested base amount from %v to %v",
+		s.Exchange,
+		s.Amount,
+		a)
+
+	s.Amount = a
+	return nil
+}
+
+// AdjustQuoteAmount will adjust the quote amount of a submit response if the
+// exchange has modified the amount. This is usually due to decimal place
+// restrictions or rounding. This will return an error if the amount is zero
+// or the submit response is nil.
+func (s *SubmitResponse) AdjustQuoteAmount(a float64) error {
+	if s == nil {
+		return errOrderSubmitResponseIsNil
+	}
+
+	if a <= 0 {
+		return errAmountIsZero
+	}
+
+	if s.QuoteAmount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "exchange %v has adjusted requested quote amount from %v to %v",
```

Comment:
Please include orderID. Less important would be asset and pair depending on exchange